### PR TITLE
[IMP] im_livechat: improve feedback messages

### PR DESCRIPTION
**Pull request by tsm-odoo**
This PR improve feedback messages:
-Visitor has left the conversation: as note
- Rating message: as note
- Rating message: Use same emojis as for customer 
- Rating reason in same message as the emoji, without writing "reason"

part of task-3332872

### CODE REVIEW

File: addons/im_livechat/models/discuss_channel.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def _close_livechat_session(self, **kwargs):
                 return
             # Notify that the visitor has left the conversation
             self.message_post(author_id=self.env.ref('base.partner_root').id,
-                              body=self._get_visitor_leave_message(**kwargs), message_type='comment', subtype_xmlid='mail.mt_comment')
+                              body=self._get_visitor_leave_message(**kwargs), message_type='comment', subtype_xmlid='mail.mt_note')
```
Comment (seb-odoo) [inline addons/im_livechat/models/discuss_channel.py:127]: `mt_note` (and the use of `subtype_xmlid` in general) was I believe limited to chatter until now, and I'd prefer to keep it that way if possible.

Can't we use message type (notification?) instead?

File: addons/im_livechat/static/src/embed/core/livechat_service.js [hunk 2]
```diff
@@ -173,16 +167,17 @@ export class LivechatService {
...
+        const emojiSrc = `${session.origin}/rating/static/src/img/rating_${
+            RATING[Object.keys(RATING).find((key) => RATING[key] === rate)]
+        }.png`;
         await this.rpc("/im_livechat/chat_post", {
             uuid,
-            message_content: sprintf(_t("Rating: %s"), RATING_TO_EMOJI[rate]),
+            message_content: sprintf(
+                _t("Rating: %s"),
+                `<img height="25" width="25" src="${emojiSrc}" alt="rating_${rate}" /> ${reason}`
+            ),
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/embed/core/livechat_service.js:178]: This is not the first occurrence (and not necessarily blocking), but it would be better if that was meta-data rather than hard-coded value.

I'm thinking maybe we could have either extra table(s)/field(s) for message meta data or a specific message type + specific data (json?) in body?

File: addons/website_livechat/models/discuss_channel.py [hunk 3]
```diff
@@ -60,10 +60,10 @@ def _get_visitor_history(self, visitor):
     def _get_visitor_leave_message(self, operator=False, cancel=False):
         name = _('The visitor') if not self.livechat_visitor_id else self.livechat_visitor_id.display_name
         if cancel:
-            message = _("""%s has started a conversation with %s. 
+            message = _("""%s has started a conversation with %s.
```
Comment (seb-odoo) [inline addons/website_livechat/models/discuss_channel.py:63]: "has started" -> "started", similar to how "has left" was changed?

File: addons/im_livechat/models/discuss_channel.py (hunk 1 above)
Comment (tsm-odoo) [reply addons/im_livechat/models/discuss_channel.py:127]: I don't think so, for the "Visitor left the conversation" it might be OK but it won't work for the rating + reason (would be very strange to have such a long notification)

File: addons/im_livechat/static/src/embed/core/livechat_service.js (hunk 2 above)
Comment (tsm-odoo) [reply addons/im_livechat/static/src/embed/core/livechat_service.js:178]: Done with CSS instead

File: addons/im_livechat/models/discuss_channel.py (hunk 1 above)
Comment (alexkuhn) [reply addons/im_livechat/models/discuss_channel.py:127]: Would be indeed best to show "Visitor left the conversation" as a notification, with `message_type="notification"`.
For rating, we can turn this one into `mt_note`.

File: addons/im_livechat/static/tests/embed/feedback_panel_tests.js [hunk 4]
```diff
@@ -57,15 +58,11 @@ QUnit.test("Feedback with rating and comment", async (assert) => {
...
-                    assert.strictEqual(
-                        args.message_content,
-                        `Rating: ${RATING_TO_EMOJI[RATING.GOOD]}`
-                    );
-                } else if (messageCount === 2) {
-                    assert.strictEqual(args.message_content, "Rating reason: Good job!");
+                if (messageCount++ === 1) {
+                    assert.ok(args.message_content.includes("Good job!"));
+                    const emojiSrc = `${session.origin}/rating/static/src/img/rating_${RATING.GOOD}.png`;
+                    assert.ok(args.message_content.includes(`src="${emojiSrc}"`));
```
Comment (alexkuhn) [inline addons/im_livechat/static/tests/embed/feedback_panel_tests.js:64]: I would remove `messageCount`. This looks irrelevant for the test, and also makes reading test harder than it should. (+ I don't like `++` and `===` combined in a single expression 🥲

File: addons/im_livechat/models/discuss_channel.py (hunk 1 above)
Comment (tsm-odoo) [reply addons/im_livechat/models/discuss_channel.py]: Juliette told me she wanted it to be a note as well, so I think we could just keep it like it?

File: addons/im_livechat/models/discuss_channel.py [hunk 5]
```diff
@@ -123,8 +123,13 @@ def _close_livechat_session(self, **kwargs):
             if not self.message_ids:
                 return
             # Notify that the visitor has left the conversation
-            self.message_post(author_id=self.env.ref('base.partner_root').id,
-                              body=self._get_visitor_leave_message(**kwargs), message_type='comment', subtype_xmlid='mail.mt_comment')
+            self.message_post(
+                author_id=self.env.ref('base.partner_root').id,
+                body=Markup('<div class="o_mail_notification o_mail_notification_hide_author">%s</div>')
```
Comment (alexkuhn) [inline addons/im_livechat/models/discuss_channel.py:128]: I would just name modifier class `o_hide_author`

**Review by alexkuhn — APPROVED** [review]
LGTM otherwise

**Review by alexkuhn — APPROVED** [review]
@robodoo override=ci/security r+
Markup is sanitized thanks to message body being a html field. `rating_url` is specified by model.
